src/session/chat/mod.rs

```diff
@@ -481,6 +488,20 @@ impl Chat {
 
     pub fn title_expression(&self) -> gtk::Expression {
         let chat_expression = gtk::ConstantExpression::new(self);
-        gtk::PropertyExpression::new(Chat::static_type(), Some(&chat_expression), "title").upcast()
+        let base_expression =
```

Comment:
To implement this properly is actually a bit more involved: we need to check if the type of the chat is "private" and in that case we should use `User::title_expression()`, else we should use regular `Chat::title` property expression. The problem is that the `Chat::type` only stores the id of the objects, but we'd need the proper objects instead to call `User::title_expression()`.

Now, if that's ok for you, I can make a separate PR to change the `Chat::type` to store full objects instead of just ids (which is something I already wanted to do), so then you could rebase your branch and implement this properly.